test/unit/insights.test.js

```diff
@@ -0,0 +1,40 @@
+describe('application insights', () => {
+  const DEFAULT_ENV = process.env
+  let applicationInsights
+
+  beforeEach(() => {
+    jest.resetModules()
+    jest.mock('applicationinsights', () => {
+      return {
+        setup: jest.fn().mockReturnThis(),
+        start: jest.fn(),
+        defaultClient: {
+          context: {
+            keys: [],
+            tags: []
+          }
+        }
+      }
+    })
+    applicationInsights = require('applicationinsights')
+    process.env = { ...DEFAULT_ENV }
+  })
```

Comment:
Is it okay for the jest.mock to be defined here?

Reply:
In this case yes as has to be after `resetModules` which we need to do between tests or the environment variables don't update properly